server/service/src/processors/mod.rs

```diff
@@ -0,0 +1,110 @@
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::sync::mpsc::{self, Receiver, Sender};
+use tokio::task::JoinHandle;
+
+use crate::service_provider::ServiceProvider;
+
+use self::transfer::requisition::ProcessRequisitionTransfersError;
+use self::transfer::shipment::ProcessShipmentTransfersError;
+use self::transfer::{
+    requisition::process_requisition_transfers, shipment::process_shipment_transfers,
+};
+
+#[cfg(test)]
+mod test_helpers;
+pub(crate) mod transfer;
+
+const NUMBER_OF_CHANNELS: usize = 30;
+
+#[derive(Clone)]
+pub struct ProcessorsTrigger {
+    requisition_transfer: Sender<()>,
+    shipment_transfer: Sender<()>,
+}
+
+pub struct Processors {
+    requisition_transfer: Receiver<()>,
+    shipment_transfer: Receiver<()>,
+}
+
+#[derive(Debug, Error)]
+enum ProcessorsError {
+    #[error("Error in shipment transfer processor ({0})")]
+    ShipmentTransfer(ProcessShipmentTransfersError),
+    #[error("Error in requisition transfer processor ({0})")]
+    RequisitionTransfer(ProcessRequisitionTransfersError),
+}
+
+impl Processors {
+    pub fn init() -> (ProcessorsTrigger, Processors) {
+        let (requisition_transfer_sender, requisition_transfer_receiver) =
+            mpsc::channel(NUMBER_OF_CHANNELS);
+
+        let (shipment_transfer_sender, shipment_transfer_receiver) =
+            mpsc::channel(NUMBER_OF_CHANNELS);
+
+        (
+            ProcessorsTrigger {
+                requisition_transfer: requisition_transfer_sender,
+                shipment_transfer: shipment_transfer_sender,
+            },
+            Processors {
+                requisition_transfer: requisition_transfer_receiver,
+                shipment_transfer: shipment_transfer_receiver,
+            },
+        )
+    }
+
+    pub fn spawn(self, service_provider: Arc<ServiceProvider>) -> JoinHandle<()> {
+        let Processors {
+            mut requisition_transfer,
+            mut shipment_transfer,
+        } = self;
+
+        tokio::spawn(async move {
+            loop {
+                let result = tokio::select! {
+                    Some(_) = requisition_transfer.recv() => {
+                        process_requisition_transfers(&service_provider).map_err(ProcessorsError::RequisitionTransfer)
+                    },
+                    Some(_) = shipment_transfer.recv() => {
+                        process_shipment_transfers(&service_provider).map_err(ProcessorsError::ShipmentTransfer)
+                    },
+                    else => break,
+                };
+
+                if let Err(error) = result {
+                    log::error!("{}", error);
```

Comment:
Just logging processor error, don't want to exit server for now, will make an issue that's specific to logging in processors for discussion

Reply:
Not yet, I only just added a checkbox for it in the epic issue, will link you when it's done (although it's probably going to be a placeholder issue for now, with description of what we currently do with logging and in what scenariors we crash the server)